Upvote topic component

jrod1032 opened the pull request.

jrod1032 committed:
getTopics function now sorts by timeStamp, newest first

jrod1032 committed:
upvote function updates data whether you previously clicked button or not

still needs to display change in vote count and state to user

jrod1032 committed:
Implement updateVoteCount function

jrod1032 committed:
Implement patch route that calls updatevotecount function

jrod1032 committed:
updateVoteCount now returns updated vote count

jrod1032 committed:
Refactor update vote count function and code cleanup

jrod1032 committed:
code cleanup on app.patch /topic route

sfkiwi reviewed (changes requested):
Do you mind moving the server request into the AppComponent

jrod1032 committed:
Refactor each component so that App component is source of truth

App component makes http patch request, and rerenders list with a GET request
Topic view has been refactored to reflect these changes as well.

jrod1032 committed:
change variable names and arguments to comply with server.

jrod1032 committed:
Patch request endpoint now takes topic id

jrod1032 committed:
uncomment getAll function
